packages/shell-api/src/error-codes.ts

```diff
@@ -0,0 +1,51 @@
+
+/**
+ * Creates a new object to use as `metadata` for `Mongosh..Error`s.
+ *
+ * @param apiDetails Additional details on the blocked API
+ */
+function blockedByDriverMetadata(apiDetails: string) {
+  return {
+    driverCaused: true,
+    api: apiDetails
+  };
+}
+
+/**
+ * @mongoshErrors
+ */
+enum ShellApiErrors {
+  /**
+   * Signals calling a method that requires sharding for a collection that is not sharded
+   * or a database that does not have sharding enabled.
+   *
+   * **Solution: Be sure to enable sharding on the database or that the collection is sharded.**
+   */
+  NoSharding = 'SHAPI-10001',
```

Comment:
SInce this and the error below are the same, maybe we can rename this to be `NotConnectedToShardedCluster` for consistency?
```suggestion
  NotConnectedToShardedCluster = 'SHAPI-10001',
```